src/main/java/vg/civcraft/mc/citadel/activity/ActivityMap.java

```diff
@@ -0,0 +1,209 @@
+package vg.civcraft.mc.citadel.activity;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import vg.civcraft.mc.citadel.Citadel;
+import vg.civcraft.mc.civmodcore.CivModCorePlugin;
+import vg.civcraft.mc.civmodcore.dao.ManagedDatasource;
+import vg.civcraft.mc.civmodcore.locations.global.WorldIDManager;
+import vg.civcraft.mc.namelayer.group.Group;
+
+public class ActivityMap {
+
+	private static final String CREATE_TABLE = "CREATE TABLE IF NOT EXISTS ctdl_activity_map ("
+			+ "group_id INT, "
+			+ "world SMALLINT, "
+			+ "x INT, "
+			+ "z INT, "
+			+ "resolution INT, "
+			+ "activity TIMESTAMP, "
+			+ "PRIMARY KEY(group_id, world, x, z, resolution))";
+	private static final String GET_ACTIVITY = "SELECT x, z, activity FROM ctdl_activity_map WHERE group_id = ? AND world = ? AND resolution = ?";
```

Comment:
Why is resolution part of the table?

Reply:
so that it can be upscaled or downscaled in the future if the resolution changes

Reply:
not necessary though

Reply:
Seems a bit redundant, especially given that there is no functionality to actually convert it. Not very important though.